### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by tafodinho**

### CODE REVIEW

File: src/components/ButtonPanel.js [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
-                <Button name="2" />
-                <Button name="3" />
-                <Button color="orange" name="+" />
-            </div>
-            <div className="group-5">
-                <Button name="0" special={true}/>
-                <Button name="." />
-                <Button color="orange" name="=" />
+class ButtonPanel extends React.Component {
+    handleClick = (btnName) => {
```
Comment (rvvergara) [inline src/components/ButtonPanel.js]: As you will observe, we are repetitively declaring `handleClick` here in `ButtonPanel` and once again in `Button`. 

`ButtonPanel` should simply pass `clickHandler` as a prop to `Button`. It is not reponsible for actually calling on the `clickHandler` method it received from `App`.

As such we don't need `handleClick` here in `ButtonPanel`. Let us simply pass `clickHandler` into `Button` component.

File: src/logic/calculate.js [hunk 2]
```diff
@@ -1,13 +1,52 @@
 import operate from './operate'
 
 const calculate = (data, btnName) => {
-    if(btnName === '+/-') {
-        data.result *= -1
+    console.log(data)
```
Comment (rvvergara) [inline src/logic/calculate.js]: Let us remove any `console.log` from our code

**Review by rvvergara — CHANGES_REQUESTED** [review]
### Hi @tafodinho,

## Git Flow

Let us follow the proper Git Flow here. I have made mention of this previously

## Linters

Let us follow the instructions on setting up a linter for our project.

On top of ALL the changes I made mention in the previous milestones we need to fix the ff:
## Issues:

**Please fix #6**

**Please fix #7**

**Please fix #8**

## Let us make sure the display changes the sign of a number when `+/-` is clicked.

Please refer to this example [Live Demo](https://ryto-react-calculator.netlify.com/) to see how our calculator should function. Don't worry about the `Ready` message at the top of the screen. Just focus on how the calculator operates and how each button should function.

## ONLY App component should be a class component.

I have made mention of this as well in the previous milestones. So please let us make sure the succeeding reviews already made this change.

## PropType validations
Let us implement this

## Style folder - why do we have redundant css and scss folders? Why don't we use just either css or scss? Also why not just use either `styles.scss` or `styles.css`? Why do we need both?

**Review by rvvergara — CHANGES_REQUESTED** [review]
### Hi @tafodinho 

Thanks for fixing the problems on the dependencies. Next time we create a react project let us use `npx create-react-app` so we are sure we are using the latest versions of the dependencies.

## We still have the following changes to make:

## Linter

Let us please follow the exact instructions laid out for this. For instance we need a `stickler.yml` file and we need stickler to be activated on this project to give us automated checks for all Pull Requests.

## Git Flow
Again the Git Flow required is still not being followed here. ALL our PR's should have `development` as the base branch. PLEASE oh please let us make sure we do this. 

### Great job fixing #6 and #7. BUT #8 is still persisting. Let us make sure our % button does its job

## Flexbox -> Our calculator needs to be flexible

We are required to use `flexbox` such that whenever we shrink the width of the browser our calculator's width also shrinks with it. Kindly see [this](https://ryto-react-calculator.netlify.com/) demo calculator to see this behaviour. When you open this demo calculator try to make the browser width smaller and see how the calculator responds.

## Failed Proptype errors

Let us fix all these
image


### Redundant folders and files inside `styles` -> Please address this. Why do we need separate `css` and `scss` folders and files? Why can't we just use either plain `css` files or `scss` files?

**Comment by tafodinho** [comment]
Hi @rvvergara the reason for the redundant folders in the styles folder is a more concise convention for using sass as i read from several post and it gives from for future expansion and clarity on the project files.

File: src/components/Button.js [hunk 3]
```diff
@@ -1,11 +1,17 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 
 const Button = (props) => {
+    const {name} = props
```
Comment (BalintMendli) [inline src/components/Button.js]: Why don't we destructure the other props as well?

**Review by BalintMendli — CHANGES_REQUESTED** [review]
Hi @tafodinho ,

Good job fixing the issues from previous reviews. The calculator works fine now.

However, we still have a Prop type error. We should pass Display a string instead of a number.

Please fix that and submit another review request.

**Review by BalintMendli — APPROVED** [review]
Hi @tafodinho ,

Everything looks good now.

### 👍 

You can merge the project.

**Pull request merged**